### PR TITLE
Feature/house search

### LINKED ISSUE #29: Support search by position, date range, and number of beds
with a single entry "house/search"

### PR DESCRIPTION
**Pull request by xingweitian**
### Your checklist for this pull request
🚨Please review the [workflow](https://github.com/new-airbnb/wiki/wiki/Workflow#introduction) before creating a pull request.

- [x] Make sure you have tested your code additions. Also the whole system can work properly in your environment.
- [x] Check if you have finished the relating issues in this pull request.
- [x] Check the commit's or even all commits' message styles matches our requested structure.
- [x] Check your code style, comments style.

#### What issue(s) does this PR aim to solve?
resolves #29 

#### What is the purpose of the changes made?
finish /house/search api.

#### What tests are done?
test by postman.

💔Thank you!

### CODE REVIEW

File: house/views.py [hunk 1]
```diff
@@ -110,3 +112,58 @@ def upload_photo(request):
...
+            "success": 1,
+            "house_list": house_list
+        })
+    else:
+        qs = House.objects.all()
+        houses = qs[:10] if len(qs) >= 10 else qs
+        return_house_list = [each.dict_it() for each in houses]
+        return JsonResponse({
+            "success": 1,
+            "homepage_recommend": homepage_recommend,
```
Comment (zhangjiangqige) [inline house/views.py]: why return this?

File: db/dbutils.py [hunk 2]
```diff
@@ -18,25 +18,61 @@ def db_connection():
...
+def geo_info_search(db, coordinate):
+    collection = db[conf["database_name"]][conf["house_location_collection"]]
+    _GeoJSON = {
+        "location": {
+            "$near": {
+                "$geometry": {
+                    "type": "Point",
+                    "coordinates": coordinate
+                },
+                "$maxDistance": 20*1000,
```
Comment (zhangjiangqige) [inline db/dbutils.py]: this should be an argument

File: db/dbutils.py [hunk 3]
```diff
@@ -18,25 +18,61 @@ def db_connection():
     return MongoClient(host=conf["database_uri"])
 
 
-def geo_info_save(db, name, place_id, coordinate):
+def geo_info_save(db, house_id, place_id, coordinate):
     collection = db[conf["database_name"]][conf["house_location_collection"]]
+    collection.ensure_index([("location", "2dsphere")])
```
Comment (zhangjiangqige) [inline db/dbutils.py]: i remember this creates index? better put in a place for initialization only once

File: house/views.py [hunk 4]
```diff
@@ -110,3 +112,58 @@ def upload_photo(request):
...
+        if not geo_search_res:
+            logger.error(
+                "Fail to search house with coordinate: {}. Something wrong with MongoDB.".format(str(coordinate)))
+            return JsonResponse({
+                "success": 0
+            }, status=500)
+        house_list = list()
+        for each in geo_search_res:
+            house = House.objects.get(pk=each["house_id"])
+            if house.number_of_beds >= num_of_beds and house.date_begin >= date_begin and house.date_end >= date_end:
```
Comment (zhangjiangqige) [inline house/views.py]: house.date_begin shoule be <= date_begin

**Review by zhangjiangqige — CHANGES_REQUESTED** [review]

File: house/views.py (hunk 1 above)
Comment (xingweitian) [reply house/views.py]: I think it will be helpful for the frontend to display the house list, since we put all the search things into the same api.

**Review by zhangjiangqige — APPROVED** [review]

**Pull request merged**